src/foundry/foundry.js/pixi/containers/canvasLayers/placeablesLayer.d.ts

```diff
@@ -268,6 +283,30 @@ declare global {
       options?: DocumentModificationContext
     ): Promise<Array<InstanceType<ConfiguredDocumentClassForName<DocumentName>>>>;
 
+    /**
+     * Get the world-transformed drop position.
+     * @returns Returns the transformed x, y co-ordinates, or false if the drag event was outside the canvas.
+     */
+    protected _canvasCoordinatesFromDrop(
+      event: DragEvent,
+      {
+        center
+      }: {
+        /**
+         * Return the co-ordinates of the center of the nearest grid element.
+         * @defaultValue `true`
+         */
+        center: boolean;
+      }
+    ): [tx: number, ty: number] | false;
+
+    /**
+     * Create a preview of this layer's object type from a world document and show its sheet so it can be finalized.
+     * @param createData - The data to create the object with.
+     * @param position   - The position to render the sheet at.
+     */
+    protected _createPreview(createData: unknown, { top, left }: { top: number; left: number }): Promise<void>; // FIXME: type createData better
```

Comment:
this should probably be done. we know the type of document the layer is responsible for (`DocumentName`), so it shouldn't be that difficult. I think this is what we want:
```ts
    protected _createPreview(
      createData: ConstructorDataType<InstanceType<ConfiguredDocumentClassForName<DocumentName>>['data']>,
      { top, left }: { top: number; left: number }
    ): Promise<void>;
 ```

Reply:
Yes you are right. For some reason I got it in my head that the createData was constructor data for layers themselves.